Helidon jar doesn't contain Automatic-Module-Name in MANIFEST.MF

## Environment Details
* Helidon Version: 2.5.1
* Helidon SE
* JDK version: JDK17
* OS: N/A
* Docker version (if applicable): N/A

----------

## Problem Description
Within Helidon jar like such:
https://repo1.maven.org/maven2/io/helidon/integrations/graal/helidon-graal-native-image-extension/2.5.1/helidon-graal-native-image-extension-2.5.1.jar
there is no presence of `Automatic-Module-Name ` property/attribute within MANIFEST.MF file. Current file is like such:
```
Manifest-Version: 1.0
Created-By: Apache Maven 3.6.3
Built-By: jenkins
Build-Jdk: 11.0.12
Implementation-Title: Helidon Integrations Graal Native Image Extension
Implementation-Version: 2.5.1
Implementation-Vendor-Id: io.helidon.integrations.graal
Implementation-Vendor: Oracle Corporation
Implementation-URL: https://helidon.io
Scm-Revision: df66d863cd0274bd899120d2d4e81aabaef64612
```

Such leads to an exception 
```
Caused by: java.lang.IllegalArgumentException: helidon.graal.native.image.extension: Invalid module name: 'native' is not a Java identifier
```
when used within Java modules.

Exception in this case is due to this list of Keywords:
https://docs.oracle.com/javase/specs/jls/se7/html/jls-3.html#jls-3.9

where `native`  is one of them.

Temporary workaround: rename file to let say `helidon-graal-native1-image-extension-2.5.1.jar`
Long term fix: introduce `Automatic-Module-Name` attribute into MANIFEST.MF

## Steps to reproduce
- Download https://repo1.maven.org/maven2/io/helidon/integrations/graal/helidon-graal-native-image-extension/2.5.1/helidon-graal-native-image-extension-2.5.1.jar into some folder
- Navigate to that folder and invoke such command:
```
java -p . --list-modules
```
- You should see such exception:
```
Error occurred during initialization of boot layer
java.lang.module.FindException: Unable to derive module descriptor for ./helidon-graal-native-image-extension-2.5.1.jar
Caused by: java.lang.IllegalArgumentException: helidon.graal.native.image.extension: Invalid module name: 'native' is not a Java identifier
```

--- a/integrations/graal/mp-native-image-extension/src/main/java/module-info.java
+++ b/integrations/graal/mp-native-image-extension/src/main/java/module-info.java
@@ -1,0 +1,28 @@
+/*
+ * Copyright (c) 2022 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * Extension for Graal VM native image to correctly build Helidon MicroProfile applications.
+ */
+module io.helidon.graal.nativeimage.mp {
+    requires jakarta.cdi;
+    requires svm;
+    requires io.helidon.graal.nativeimage;
+    requires jakarta.json;
+    requires org.graalvm.sdk;
+    requires weld.core.impl;
+
+    exports io.helidon.integrations.graal.mp.nativeimage.extension;
+}

--- a/integrations/graal/mp-native-image-extension/src/main/java/module-info.java
+++ b/integrations/graal/mp-native-image-extension/src/main/java/module-info.java
@@ -17,12 +17,13 @@
  * Extension for Graal VM native image to correctly build Helidon MicroProfile applications.
  */
 module io.helidon.graal.nativeimage.mp {
-    requires jakarta.cdi;
     requires svm;
     requires io.helidon.graal.nativeimage;
-    requires jakarta.json;
     requires org.graalvm.sdk;
     requires weld.core.impl;
+    requires jakarta.enterprise.cdi.api;
+    requires java.json;
+    requires jboss.classfilewriter;
 
     exports io.helidon.integrations.graal.mp.nativeimage.extension;
 }

--- a/integrations/graal/native-image-extension/src/main/java/module-info.java
+++ b/integrations/graal/native-image-extension/src/main/java/module-info.java
@@ -1,0 +1,28 @@
+/*
+ * Copyright (c) 2022 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * Extension for Graal VM native image to correctly build Helidon applications.
+ */
+module io.helidon.graal.nativeimage {
+    requires io.helidon.config;
+    requires io.github.classgraph;
+    requires io.helidon.config.mp;
+    requires svm;
+    requires org.graalvm.sdk;
+    requires jakarta.json;
+
+    exports io.helidon.integrations.graal.nativeimage.extension;
+}

--- a/integrations/graal/native-image-extension/src/main/java/module-info.java
+++ b/integrations/graal/native-image-extension/src/main/java/module-info.java
@@ -22,7 +22,7 @@ module io.helidon.graal.nativeimage {
     requires io.helidon.config.mp;
     requires svm;
     requires org.graalvm.sdk;
-    requires jakarta.json;
+    requires java.json;
 
     exports io.helidon.integrations.graal.nativeimage.extension;
 }

--- a/tests/integration/native-image/mp-1/src/main/resources/META-INF/microprofile-config.properties
+++ b/tests/integration/native-image/mp-1/src/main/resources/META-INF/microprofile-config.properties
@@ -1,5 +1,5 @@
 #
-# Copyright (c) 2019, 2020 Oracle and/or its affiliates. All rights reserved.
+# Copyright (c) 2019, 2022 Oracle and/or its affiliates.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.

